Lib/shutil.py

```diff
@@ -170,6 +171,35 @@ def _copyxattr(src, dst, *, follow_symlinks=True):
     def _copyxattr(*args, **kwargs):
         pass
 
+if sys.platform == 'darwin':
+    # As of macOS 10.10, the `copyfile` API changed to not copy certain flags
+    mac_ver = tuple(int(x) for x in platform.mac_ver()[0].split('.'))
+    if mac_ver >= (10, 10, 0):
+        def _fix_flags(dst, flags, follow_symlinks):
+            """Perform any modifications to `flags` before they can be applied
+            to `dst` with `chflags()`.
+
+            """
+
+            # From xnu's bsd/sys/stat.h
+            UF_TRACKED    = 0x00000040
+            SF_RESTRICTED = 0x00080000
+
+            # Based on copyfile's copyfile_stat()
+            omit_flags = (UF_TRACKED | SF_RESTRICTED)
+            add_flags  = 0
+
+            # If the kernel automatically put SF_RESTRICTED on the destination
+            # already, we don't want to clear it
+            st = os.stat(dst, follow_symlinks=follow_symlinks)
+            add_flags |= (st.st_flags & SF_RESTRICTED)
+
+            return (flags & ~omit_flags) | add_flags
+
+if '_fix_flags' not in vars():
```

Comment:
Use "else:" instead

Reply:
If I used `else:`, it would make the code more complicated. If it matched with `if sys.platform == 'darwin'`, then `_fix_flags` would not be defined for old versions of macOS. If I matched it with `if mac_ver >= (10, 10, 0)`, then it would not be defined for other platforms. If I wanted to do both, I'd have to repeat code. This seemed the least complex.